Update README.md for replacing `--sdk-auth` with `--json-auth` (#342)
Azure/login: README.md

For using any credentials like Azure Service Principal, Publish Profile etc add them as [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) in the GitHub repository and then use them in the workflow.

> Follow the following steps to configure Azure Service Principal with a secret in the scope of `resource-group` as the role of `contributor`:

Reviewer: ```suggestion
Follow the following steps to configure Azure Service Principal with a secret at the scope of `{resource-group}` as the role of `contributor`:
```

```

- Now in the workflow file in your branch: `.github/workflows/workflow.yml` replace the secret in Azure login action with your secret (Refer to the example above)

> - Note: The above `az ad sp create-for-rbac` command will give you the `--sdk-auth` deprecation warning. As we are working with CLI for this deprecation process, we strongly recommend users to use this `--sdk-auth` flag as the result dictionary output changes and not accepted by login action if `--sdk-auth` is not used.

Reviewer: There is even a typo!

> **and** not accepted

**and** should be **are**.

> - Note: The parameter `--json-auth` outputs the result dictionary accepted by the login action, accessible in Azure CLI versions `>= 2.51.0`. Versions prior to this use `--sdk-auth` with a deprecation warning.

Reviewer: Azure CLI uses the word "argument", instead of "parameter":

```suggestion
- Note: The argument `--json-auth` outputs the result dictionary accepted by the login action, accessible in Azure CLI versions `>= 2.51.0`. Versions prior to this use `--sdk-auth` with a deprecation warning.
```

> - Note: The parameter `--json-auth` outputs the result dictionary accepted by the login action, accessible in Azure CLI versions `>= 2.51.0`. Versions prior to this use `--sdk-auth` with a deprecation warning.

Reviewer: Azure CLI uses the word "argument", instead of "parameter":

```suggestion
- Note: Starting from Azure CLI versions 2.51.0, the argument `--json-auth` outputs the result dictionary accepted by the login action. It is equivalent to `--sdk-auth` in older versions of Azure CLI, but `--sdk-auth` shows deprecation warning.
```
